clustermesh-apiserver/clustermesh/converters.go

```
@@ -126,27 +136,52 @@ func (nc *CiliumNodeConverter) Convert(event resource.Event[*cilium_api_v2.Ciliu
+		Enabled:               true,
+		Resource:              "CiliumIdentity",
+		Prefix:                path.Join(identityCache.IdentitiesPath, "id"),
+		StoreOpts:             []store.WSSOpt{store.WSSWithSyncedKeyOverride(identityCache.IdentitiesPath)},
+		NamespaceSyncRequired: true, // Only global namespaces should be synced
 	}
 }
 
 // CiliumIdentityConverter implements Converter[*cilium_api_v2.CiliumIdentity]
-type CiliumIdentityConverter struct{ logger *slog.Logger }
+type CiliumIdentityConverter struct {
+	logger           *slog.Logger
+	namespaceManager cmnamespace.Manager
+}
 
-func newCiliumIdentityConverter(logger *slog.Logger) Converter[*cilium_api_v2.CiliumIdentity] {
-	return &CiliumIdentityConverter{logger: logger}
+func newCiliumIdentityConverter(logger *slog.Logger, namespaceManager cmnamespace.Manager) Converter[*cilium_api_v2.CiliumIdentity] {
+	return &CiliumIdentityConverter{
+		logger:           logger.With(logfields.LogSubsys, "CiliumIdentityConverter"),
+		namespaceManager: namespaceManager,
+	}
 }
 
 func (ic *CiliumIdentityConverter) Convert(event resource.Event[*cilium_api_v2.CiliumIdentity]) (upserts iter.Seq[store.Key], deletes iter.Seq[store.NamedKey]) {
+	scoppedLogger := ic.logger.With(logfields.Identity, event.Key.Name)
+
 	if event.Kind == resource.Delete {
 		key := store.NewKVPair(event.Key.Name, "")
 		return noneIter[store.Key], singleIter[store.NamedKey](key)
 	}
 
 	identity := event.Object
+	// Only upsert identities that are in global namespaces.
+	namespace, exists := identity.SecurityLabels[podPrefixLbl]
+	if !exists {
+		scoppedLogger.Error("Security label does not exist for identity", logfields.Label, podPrefixLbl)
+		return noneIter[store.Key], noneIter[store.NamedKey]
+	}
+	isGlobal, err := ic.namespaceManager.IsGlobalNamespaceByName(namespace)
+	if err != nil {
+		scoppedLogger.Error("Error checking if namespace is global", logfields.Error, err)
+		return noneIter[store.Key], noneIter[store.NamedKey]
+	}
```

Ignoring the error here is not correct, because it is possible that the store hasn't seen the namespace yet, and that would mean that the resource is incorrectly not propagated. 

I think there are two main possibilities to address this:
1. Let these methods return an error, which can be then passed to `event.Done` so that the event is re-enqueued and eventually processed again;
2. Move the logic to perform filtering directly to the synchronizer. For instance, once may introduce a new interface (e.g., `Namespacer`) that knows how to retrieve the namespace from any given type of resource, and then have the logic to perform filtering directly there. I'd personally suggest to explore this alternative, as it would have the advantage of keeping the converters agnostic, and grouping all the filtering logic in a single place (while it is now spread between converters and the synchronizer).

---

Makes sense. Moved filtering to synchronizer.